### PR TITLE
RavenDB-16450 MultiGetCommand - handle cache & respond of not found

### PR DESCRIPTION
**Pull request by haludi**
https://issues.hibernatingrhinos.com/issue/RavenDB-16450

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/DocumentHandler.cs [hunk 1]
```diff
@@ -222,7 +223,9 @@ private async Task GetDocumentsByIdAsync(DocumentsOperationContext context, Micr
                 }
                 if (document == null && ids.Count == 1)
                 {
-                    HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                    HttpContext.Response.StatusCode = GetStringFromHeaders("If-None-Match") == HttpCache.NotFoundResponse
```
Comment (haludi) [inline src/Raven.Server/Documents/Handlers/DocumentHandler.cs:226]: When requesting multiple documents that do not exist we respond with `NotModified` (if the previous response was the same) but when requesting only one we respond with `NotFound`.

File: test/SlowTests/Bugs/Caching/UseCachingInLazyTests.cs [hunk 2]
```diff
@@ -422,6 +421,66 @@ await using (var bulk = store.BulkInsert())
...
+                using (var session = store.OpenAsyncSession())
+                {
+                    //Add "NotExistDocId" to cache
+                    await session.Advanced.Lazily.LoadAsync<TestObj>(notExistDocId).Value;
+                    Assert.Equal(0, session.Advanced.NumberOfRequests);
+                }
+            }
+
+#if DEBUG
+            await Task.Delay(10000);
```
Comment (haludi) [inline test/SlowTests/Bugs/Caching/UseCachingInLazyTests.cs]: I found it the best way to test it without modifying the production code for the test and without relying too much on the internal implementation.

File: src/Raven.Client/Http/HttpCache.cs [hunk 3]
```diff
@@ -14,6 +14,8 @@ namespace Raven.Client.Http
 {
     public class HttpCache : IDisposable, ILowMemoryHandler
     {
+        public const string NotFoundResponse = "404 Response";
```
Comment (ppekrol) [inline src/Raven.Client/Http/HttpCache.cs]: private or internal

File: test/SlowTests/Bugs/Caching/UseCachingInLazyTests.cs (hunk 2 above)
Comment (ppekrol) [reply test/SlowTests/Bugs/Caching/UseCachingInLazyTests.cs]: Let's remove this. The whole test suite will die anyway.

**Pull request merged**